### PR TITLE
Enhancement/5320 add integration test escape secrets in policy

### LINKED ISSUE #5320: E2E Test: escaping secrets in policy
Add an E2E test for https://github.com/elastic/kibana/issues/188377, wherein the test should create a policy containing the custom logs integration with a custom configuration like so:
```
custom.key: $$$$
```

The test should enroll Agent in Fleet using the created policy and ensure that the configuration sent to the relevant Beat contains `custom.key: $$$$` as well.

### PR DESCRIPTION
**Pull request by kaanyalti**
- Enhancement

## What does this PR do?

Adds integration test for testing escaping secrets in the agent config. This test case replicates the steps outlined in the [following issue](https://github.com/elastic/kibana/issues/188377).

## Why is it important?

We currently don't have integration tests covering this use case

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [x] I have added an integration test or an E2E test

## How to test this PR locally

Run ` GOTEST_FLAGS='-test.run ^TestEscapingSecretsInPolicy$' mage integration:test` for the platforms and packages that you want

## Related issues

- Relates [188377](https://github.com/elastic/kibana/issues/188377)
- Closes #5320

### CODE REVIEW

File: changelog/fragments/1731292950-added-integration-tests-for-escaping-secrets-in-policy.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: enhancement
+
+# Change summary; a 80ish characters long description of the change.
+summary: added integration tests for escaping secrets in policy
```
[the remark this answers is not in the document]
Comment (kaanyalti) [reply changelog/fragments/1731292950-added-integration-tests-for-escaping-secrets-in-policy.yaml:14]: Sounds good, removing it

[the remark this answers is not in the document]
Comment (kaanyalti) [reply changelog/fragments/1731292950-added-integration-tests-for-escaping-secrets-in-policy.yaml]: fragment check is failing when I remove the changelog entry; though I presume there is a way to bypass it, is that correct? @rdner

[the remark this answers is not in the document]
Comment (rdner) [reply changelog/fragments/1731292950-added-integration-tests-for-escaping-secrets-in-policy.yaml]: @kaanyalti yeah, it's the `skip-changelog` label. I already set it.

**Review by blakerouse — APPROVED** [review]
Looks good, covers the test requirements.

**Review by rdner — APPROVED** [review]

**Review by swiatekm — APPROVED** [review]

**Pull request merged**